ui/src/app/shared/components/dropdown/dropdown.tsx

```diff
@@ -19,112 +17,84 @@ export interface DropDownState {
 
 require('./dropdown.scss');
 
-const dropDownOpened = new BehaviorSubject<DropDown>(null);
+export function DropDown({isMenu, anchor, children, qeId}: DropDownProps) {
+    const [opened, setOpened] = useState(false);
+    const [left, setLeft] = useState(0);
+    const [top, setTop] = useState(0);
+    const anchorRef = useRef<HTMLDivElement>(null);
+    const contentRef = useRef<HTMLDivElement>(null);
+    const anchorEl = anchorRef.current;
+    const contentEl = contentRef.current;
 
-export class DropDown extends React.Component<DropDownProps, DropDownState> {
-    private el: HTMLDivElement;
-    private content: HTMLDivElement;
-    private subscriptions: Subscription[];
+    function refreshState() {
+        const anchorHeight = anchorEl.offsetHeight + 2;
+        const {top: anchorTop, left: anchorLeft} = anchorEl.getBoundingClientRect();
+        const newState = {top, left, opened};
 
-    constructor(props: DropDownProps) {
-        super(props);
-        this.state = {opened: false, left: 0, top: 0};
-    }
-
-    public render() {
-        let children: React.ReactNode = null;
-        if (typeof this.props.children === 'function') {
-            if (this.state.opened) {
-                const fun = this.props.children as () => React.ReactNode;
-                children = fun();
-            }
-        } else {
-            children = this.props.children as React.ReactNode;
-        }
-
-        return (
-            <div className='argo-dropdown' ref={el => (this.el = el)}>
-                <div
-                    qe-id={this.props.qeId}
-                    className='argo-dropdown__anchor'
-                    onClick={event => {
-                        this.open();
-                        event.stopPropagation();
-                    }}>
-                    <this.props.anchor />
-                </div>
-                {ReactDOM.createPortal(
-                    <div
-                        className={classNames('argo-dropdown__content', {'opened': this.state.opened, 'is-menu': this.props.isMenu})}
-                        style={{top: this.state.top, left: this.state.left}}
-                        ref={el => (this.content = el)}>
-                        {children}
-                    </div>,
-                    document.body
-                )}
-            </div>
-        );
-    }
-
-    public componentWillMount() {
-        this.subscriptions = [
-            merge(
-                dropDownOpened.pipe(filter(dropdown => dropdown !== this)),
-                fromEvent(document, 'click').pipe(
-                    filter((event: Event) => {
-                        return this.content && this.state.opened && !this.content.contains(event.target as Node) && !this.el.contains(event.target as Node);
-                    })
-                )
-            ).subscribe(() => {
-                this.close();
-            }),
-            fromEvent(document, 'scroll', {capture: true}).subscribe(() => {
```

Comment:
<img width="1406" alt="image" src="https://github.com/argoproj/argo-workflows/assets/100748721/5f026509-62c4-4c77-9fd4-237e33069b09">
- This scroll event simply moves this dropdown position to a new location when the scroll event occurs. However, this doesn't seem to be working correctly. 
- This image is a test of the scroll event on the master branch The filter moves up when the screen is small and works fine when the screen is large, so I'm not sure what the need for the event is 😅 Let me know what you think and I'll try to incorporate it, thanks 🙇‍♂️

Reply:
Ok, I didn't realize that scroll code was broken, thanks for bringing that up!
I tested myself, and yes it didn't seem very effective or useful. Maybe this logic predates the CSS being `fixed` or something? 🤷 

So I think your initial removal was the correct move forward, let's remove it again. Sorry for the re-work!

For future reference, when you change something that may be unexpected to a reviewer, a common practice is to either mention it in your PR description -- I use a "Notes for Reviewers" section occasionally -- or comment on your own PR and leave an in-line comment explaining why it was changed. You can kind of predict a question like this and answer it ahead of time then, making the review more efficient 🙂
It also saves that information for posterity in the comments instead of just in your brain or a reviewer's brain. So that, for example, if a future reader wonders why this was removed, they can just look at this comment and know -- no need to guess!

Reply:
- I wasn't aware of that... Am I correct in saying that I should either get a link to the comment with the explanation of why the scroll event was removed and create a separate comment, or create a section in the description part of the PR "Notes for Reviewers" and write it down?
- Thanks for always teaching me so much. Come to think of it, I don't think it would be easy to find this comment without a separate link 🤣.

Reply:
I think adding a "Notes for Reviewers" with a link to this comment thread would be sufficiently helpful for posterity 🙂